### PR TITLE
Pull/k32w adaptation 15.4

### PR DESCRIPTION
**Pull request by doru91**
This pull-request adds the 15.4 adaptation layer for NXP K32W platform (https://www.nxp.com/products/wireless/thread/k32w061-41-high-performance-secure-and-ultra-low-power-mcu-for-zigbeethread-and-bluetooth-le-5-0-with-built-in-nfc-option:K32W061_41).

Most of the code is taken from OpenWeave and the namespace was changed to Project CHIP.

In the following PRs will add the changes required for the build system and also a demo lock app.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
@rwalker-apple ?

**Review by andy31415 — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

File: src/platform/K32W/ConnectivityManagerImpl.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#else
+#include <platform/internal/GenericConnectivityManagerImpl_NoThread.h>
+#endif
+#include <platform/internal/GenericConnectivityManagerImpl_NoTunnel.h>
+#include <platform/internal/GenericConnectivityManagerImpl_NoWiFi.h>
+#include <support/FlagUtils.hpp>
+
+namespace nl {
+namespace Inet {
+class IPAddress;
```
Comment (rwalker-apple) [inline src/platform/K32W/ConnectivityManagerImpl.h:41]: this appears not to be needed by anything in this header.  why is it here?

File: src/platform/K32W/Logging.cpp [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+{
+    va_list v;
+
+    (void) aLogLevel;
+    (void) aLogRegion;
+
+    va_start(v, aFormat);
+
+#if K32W_LOG_ENABLED
+    char formattedMsg[CHIP_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE - 1] = { 0 };
```
Comment (rwalker-apple) [inline src/platform/K32W/Logging.cpp:226]: this near copy of the code above in LogV() is a little unfortunate

File: src/platform/K32W/Logging.cpp [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+            K32WWriteBlocking((const uint8_t *) formattedMsg, strlen(formattedMsg));
+        }
+    }
+
+    // Let the application know that a log message has been emitted.
+    DeviceLayer::OnLogOutput();
+
+#endif // NRF_LOG_ENABLED
+}
```
Comment (dleach02) [inline src/platform/K32W/Logging.cpp:170]: Should probably remove the NRF names that are not aligned with the K32W names

Comment (doru91) [reply src/platform/K32W/Logging.cpp:170]: Will remove the comment.

**Pull request merged**